format-delimited/src/main/java/io/cdap/plugin/format/delimited/input/DelimitedInputFormatProvider.java

```diff
@@ -96,4 +109,79 @@ private static PluginClass getPluginClass() {
     return new PluginClass(ValidatingInputFormat.PLUGIN_TYPE, NAME, DESC, DelimitedInputFormatProvider.class.getName(),
                            "conf", properties);
   }
+
+  @Nullable
```

Comment:
when can schema be null?

Reply:
I'm not sure.  I am referring to similar function declared in [avro-format](https://github.com/cdapio/hydrator-plugins/blob/aedb88784e5e9f5dbd9b87db030ef7f8c61d184c/format-avro/src/main/java/io/cdap/plugin/format/avro/input/AvroInputFormatProvider.java#L89-L107) and [format-parquet](https://github.com/cdapio/hydrator-plugins/blob/aedb88784e5e9f5dbd9b87db030ef7f8c61d184c/format-parquet/src/main/java/io/cdap/plugin/format/parquet/input/ParquetInputFormatProvider.java#L72-L83). I think it may be a mistake here and in the files I'm referring. Can you have a look and confirm?

Reply:
When schema contains macros, super.getSchema() should return null.  I am referring to similar function declared in [avro-format](https://github.com/cdapio/hydrator-plugins/blob/aedb88784e5e9f5dbd9b87db030ef7f8c61d184c/format-avro/src/main/java/io/cdap/plugin/format/avro/input/AvroInputFormatProvider.java#L89-L107) and [format-parquet](https://github.com/cdapio/hydrator-plugins/blob/aedb88784e5e9f5dbd9b87db030ef7f8c61d184c/format-parquet/src/main/java/io/cdap/plugin/format/parquet/input/ParquetInputFormatProvider.java#L72-L83).
Before this change, [the function in DelimitedConfig.java](https://github.com/cdapio/hydrator-plugins/blob/develop/format-delimited/src/main/java/io/cdap/plugin/format/delimited/input/DelimitedConfig.java#L85-L86) used to return null directly.

Reply:
Change discarded